aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/Translator.java

```diff
@@ -178,15 +173,31 @@ static ListDomainsRequest translateToListRequest(final String nextToken) {
    * @param awsResponse the aws service describe resource response
    * @return list of resource models
    */
-  static List<ResourceModel> translateFromListRequest(final ListDomainsResponse awsResponse) {
+  static List<ResourceModel> translateFromListRequest(
+      final ListDomainsResponse awsResponse, final ResourceHandlerRequest<ResourceModel> request
+  ) {
+
     return streamOfOrEmpty(awsResponse.domains())
         .map(domain -> ResourceModel.builder()
+            .arn(buildArn("us-west-2", domain.owner(), domain.name()))
             // TODO change domainName to arn when CodeArtifactClient populates arn in the ListDomainsResponse
-            .domainName(domain.name())
             .build())
         .collect(Collectors.toList());
   }
 
+  // TODO change domainName to arn when CodeArtifactClient populates arn in the ListDomainsResponse
+  private static String buildArn(String region, String domainOwner, String domainName) {
+    final String partition = "aws";
+    if (region.contains("us-gov-")) {
+      partition.concat("-us-gov");
+    }
+
+    if (region.contains("cn-")) {
+      partition.concat("-cn");
+    }
```

Comment:
I get that this is temporary until we can get the ListDomains to return the Arns, but i am surprised at the very least that there isnt some magic "give me my partition" in the CFN toolkit here. I know if you create a Region object (from sddv2) you can get the partition.